resources/type-system/flow-analysis.md

```diff
@@ -15,7 +15,46 @@ reachability analysis.
 
 ## Motivation
 
-TODO
+The Dart language spec requires type promotion analysis to determine when
+a local variable is known to have a more specific type than its declared type.
+This is currently performed by the front end and the analyzer which perform additional
```

Comment:
This is not really accurate--the "type propagation" phase that used to exist in the analyzer was removed with Dart 2.0.  At the moment the analyzer and the front end both implement the same type promotion rules, which match the spec.

Reply:
Ah! Missed that. Reworded.